src/main/java/de/srendi/advancedperipherals/common/addons/refinedstorage/RsApi.java

```diff
@@ -0,0 +1,679 @@
+                        }
+                    }
+                }
+            } else {
+                inputMatch = true;
+            }
+
+            if (outputFilter != null) {
+                for (ResourceAmount output : pattern.layout().outputs()) {
+                    if (outputFilter.testRS(output)) {
+                        outputMatch = true;
+                        break;
+                    }
+                }
+            } else {
+                outputMatch = true;
+            }
+
+            if (inputMatch && outputMatch)
+                return Pair.of(pattern, null);
+        }
+
+        return Pair.of(null, "NO_PATTERN_FOUND");
+    }
+
+    public static long getTotalStorage(Network network, StorageType type) {
+        long total = 0;
+
+        GraphNetworkComponent graphNetworkComponent = network.getComponent(GraphNetworkComponent.class);
+        for (InWorldNetworkNodeContainer nodeContainer : graphNetworkComponent.getContainers(InWorldNetworkNodeContainer.class)) {
+            if (nodeContainer.getNode() instanceof StorageNetworkNode storageNetworkNode) {
+                CompositeStorage storage = (CompositeStorage) storageNetworkNode.getStorage();
+                for (Storage disk : storage.getSources()) {
+                    if (!(disk instanceof StateTrackedStorage stateTrackedStorage))
+                        continue;
+
+                    if (stateTrackedStorage.getDelegate() instanceof SerializableStorage serializableStorage) {
+                        if (serializableStorage.getType() != type)
+                            continue;
+                    }
+
+                    total += stateTrackedStorage.getCapacity();
+                }
+            }
+        }
+        return total;
+    }
+
+    public static long getUsedStorage(Network network, StorageType type) {
+        long used = 0;
+
+        GraphNetworkComponent graphNetworkComponent = network.getComponent(GraphNetworkComponent.class);
+        for (InWorldNetworkNodeContainer nodeContainer : graphNetworkComponent.getContainers(InWorldNetworkNodeContainer.class)) {
+            if (nodeContainer.getNode() instanceof StorageNetworkNode storageNetworkNode) {
+                CompositeStorage storage = (CompositeStorage) storageNetworkNode.getStorage();
+                for (Storage disk : storage.getSources()) {
+                    if (!(disk instanceof StateTrackedStorage stateTrackedStorage))
+                        continue;
+
+                    if (stateTrackedStorage.getDelegate() instanceof SerializableStorage serializableStorage) {
+                        if (serializableStorage.getType() != type)
+                            continue;
+                    }
+
+                    used += stateTrackedStorage.getStored();
+                }
+            }
+        }
+        return used;
+    }
+
+    public static long getTotalExternStorage(Network network, StorageType type) {
```

Comment:
Extern -> External

same for belows

Reply:
do you think that should also be for lua functions?

Reply:
yeah sure